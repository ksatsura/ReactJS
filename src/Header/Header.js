//imports
import React from 'react';
import SearchHeader from '../SearchHeader/SearchHeader';
import AssetHeader from '../AssetHeader/AssetHeader';
import '../style.css';

export const Header = (props) => {

    const { isAssetClicked } = props;

    return (
        <div>
            { isAssetClicked ? <AssetHeader /> : <SearchHeader /> }
        </div>
    );
};

export default Header;