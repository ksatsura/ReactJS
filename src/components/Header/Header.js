//imports
import React from 'react';
import SearchHeader from '../SearchHeader/SearchHeader';
import FilmHeader from '../FilmHeader/FilmHeader';
import { Route, Switch } from 'react-router-dom';
import '../style.css';

export const Header = () => {

    return (
        <div>
            <Switch>
                <Route path='/film/:id' render={() => <FilmHeader /> } /> 
                <Route path='/' render={ () => <SearchHeader /> } /> 
                <Route path='/search/:value' render={() => <SearchHeader /> } />
            </Switch>
        </div>
    );
};

export default Header;