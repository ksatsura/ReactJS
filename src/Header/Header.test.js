import React from 'react';
import Header from './Header.js';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';

describe('Header component', () => {

    beforeAll(() => {
        jest.mock('../SearchArea/SearchArea.js', () => 'searchArea');
        jest.mock('../CompanyTitle/CompanyTitle.js', () => 'companyTitle');
    });

    it('should be render correctly', () => {
        const component = shallow(<Header />);
        
        expect(component).toMatchSnapshot();
    });
});