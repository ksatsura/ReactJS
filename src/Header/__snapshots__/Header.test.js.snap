// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be render correctly 1`] = `ShallowWrapper {}`;
